packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -81,6 +82,13 @@ class FlutterCommandRunner extends CommandRunner {
             'dependency_overrides for $kFlutterEnginePackageName, if any, or, failing that, tries to guess at the location\n'
             'based on the value of the --flutter-root option.');
 
+    argParser.addOption('engine-build',
+        hide: !verboseHelp,
+        help:
+            'Name of a build output within the engine out directory, if you are building Flutter locally.\n'
+            'Use this to select a specific version of the engine if you have built multiple engine targets.\n'
+            'This path is relative to --engine-src-path/out.');
+
     argParser.addOption('host-debug-build-path',
         hide: !verboseHelp,
         help:
```

Comment:
This is a step toward eliminating BuildConfigurations and cleaning up how we locate artifacts used during the build process.

These other flags will disappear when BuildConfigurations are gone.